pkg/compactor/shard_aware_deduplicate_filter.go

```diff
@@ -200,24 +202,31 @@ func (b *blockWithSuccessors) isIncludedIn(other *blockWithSuccessors) bool {
 	return true
 }
 
-// addSuccessorIfPossible adds successor to this block, but only if this block is included in the other block.
-// Successor is not added directly to this block, but correct place in the tree is found for it first.
+// addSuccessorIfPossible adds the given block as a successor of this block, or of a successor of this block
+// Successor is added in the correct place in the tree of successors of this block.
 // Returns true, if other block was added as successor (somewhere in the tree), false otherwise.
-//
-// Since different subtrees overlap, we need to handle the case when this block and other block are the same pointer.
 func (b *blockWithSuccessors) addSuccessorIfPossible(other *blockWithSuccessors) bool {
+	// if this and other block are the same pointer,
```

Comment:
I think the article is missing:
```suggestion
	// if this and the other block are the same pointer,
```
Is this comment necessary though? The code looks self explanatory to me.